docs/literate/src/files/first_steps/create_first_setup.jl

```diff
@@ -0,0 +1,260 @@
+# ```
+# This function must take the state variable, the spatial coordinates, the time and the
+# equation itself as arguments and returns the source term as a static vector `SVector`.
+
+function source_term_exp_sinpi(u, x, t, equations::LinearScalarAdvectionEquation2D)
+    scalar = - 2 * exp(-t) * sinpi(2*(x[1] - t)) * sinpi(2*(x[2] - t))
+    return SVector(scalar)
+end
+
+# Now we are collecting all the information that will be needed to define a spatial discretization,
+# which leaves us with an ODE problem in time with a span from 0.0 to 1.0.
+# This approach is commonly referred to as the method of lines. 
+
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver;
+                                    source_terms = source_term_exp_sinpi)
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan);
+
+# At this point, our problem is defined. We will use the `solve` function defined in
+# [OrdinaryDiffEq.jl](https://github.com/SciML/OrdinaryDiffEq.jl) to get the solution.
+# OrdinaryDiffEq.jl gives us the ability to customize the solver
+# using callbacks without actually modifying it. Trixi.jl already has some implemented
+# [Callbacks](@ref callbacks-id). The most widely used callbacks in Trixi.jl are
+# [step control callbacks](https://docs.sciml.ai/DiffEqCallbacks/stable/step_control/) that are
+# activated at the end of each time step to perform some actions, e.g. to print statistics.
+# We will show you how to use some of the common callbacks.
+
+# To print a summary of the simulation setup at the beginning
+# and to reset timers we use the [`SummaryCallback`](@ref).
+# When the returned callback is executed directly, the current timer values are shown.
+
+summary_callback = SummaryCallback()
+
+# We also want to analyze the current state of the solution in regular intervals.
+# The [`AnalysisCallback`](@ref) outputs some useful statistical information during the solving process
+# every `interval` time steps.
+
+analysis_callback = AnalysisCallback(semi, interval = 5)
+
+# It is also possible to control the time step size using the [`StepsizeCallback`](@ref) if the time
+# integration method isn't adaptive itself. To get more details, look at
+# [CFL based step size control](@ref CFL-based-step-size-control).
+
+stepsize_callback = StepsizeCallback(cfl = 1.6)
+
+# To save the current solution in regular intervals we use the [`SaveSolutionCallback`](@ref).
+# We would like to save the initial and final solutions as well. The data
+# will be saved as HDF5 files located in the `out` folder. Afterwards it is possible to visualize
+# a solution from saved files using Trixi2Vtk.jl and ParaView, which is described below in the
+# section [Visualize the solution](@ref Visualize-the-solution).
+
+save_solution = SaveSolutionCallback(interval = 5,
+                                     save_initial_solution = true,
+                                     save_final_solution = true)
```

Comment:
It might be worth mentioning here that an alternative strategy to print solution files at a fixed time interval would use the syntax
```julia
save_solution = SaveSolutionCallback(dt = 0.1,
                                     save_initial_solution = true,
                                     save_final_solution = true)
```

Reply:
Thank you for the review!
Good point, I added it.